wpimath/src/main/java/edu/wpi/first/math/apriltag/AprilTagFieldLayout.java

```diff
@@ -0,0 +1,34 @@
+package edu.wpi.first.math.apriltag;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AprilTagFieldLayout {
+    private final Map<Integer, Pose3d> m_tags = new HashMap<>();
+    private boolean m_mirror = false;
+
+    public AprilTagFieldLayout(Map<Integer, Pose3d> tags) {
+        // To ensure the underlying semantics don't change with what kind of map is passed in
+        m_tags.putAll(tags);
+    }
+
+    public void setShouldMirror(boolean mirror) {
```

Comment:
wpimath doesn't have access to that class.

Reply:
Then I think we should put the AprilTagFieldLayout class in wpilib instead of wpimath. I also suggest making the JSON filename a constructor parameter to users don't need to deal with the utility class.